actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -139,6 +141,33 @@ def negotiate_mime(order)
         order.include?(Mime::ALL) ? format : nil
       end
 
+      class VariantInquirer # :nodoc:
+        delegate :==, :each, to: :variants
+
+        def initialize(variants)
+          @variants = variants
+        end
+
+        def any?(*candidates)
+          (variants & candidates).any?
+        end
+
+        def to_ary
```

Comment:
This allows `request.variant` to be compared to other arrays, since Array comparison methods like #==, #&, and #|  call `to_ary` on the other array:

```
request.variant = :phone
# => #<VariantInquirer @variants=[:phone]>

[:phone] == request.variant
# => true
```

Existing framework code [relies on this behavior](https://github.com/rails/rails/blob/676d4ae787254e7842834e31204e5b7350261bbd/actionpack/lib/action_controller/metal/mime_responds.rb#L293) in a few places.